app/views/users/show.html.erb

```diff
@@ -1,5 +1,8 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
+  <%unless (@confirmed_friends + @pending_friends).include?(@user.id) || @user.id == current_user.id%>
+    <%= button_to 'invite', invite_path, method: :post, params: { friend_id: @user.id } %>
+  <% end %>
```

Comment:
We called that logic into views.
Please, use **views helpers** to handle that.

Don't forget to handle it in other views like `index.html.erb`